### PR TITLE
Ikasan 1780 ldap synchronisation v2

### PR DESCRIPTION
**Pull request by mick-stewart73**

### CODE REVIEW

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/security/schedule/LdapDirectorySynchronisationSchedulerService.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                " of the dashboard database not yet being created.");
+            return;
+        }
+
+        authenticationMethods.forEach(authenticationMethod -> {
+            if (authenticationMethod.isScheduled()) {
+                LdapDirectorySynchronisationJob job = new LdapDirectorySynchronisationJob(authenticationMethod,
+                    this.ldapService, this.securityService);
+                JobDetail jobDetail = this.scheduledJobFactory.createJobDetail
+                    (job, BusinessStreamNotificationJob.class, job.getJobName(), "scheduled-ldap");
```
Comment (andrzej-majewski) [inline ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/security/schedule/LdapDirectorySynchronisationSchedulerService.java]: This line does not look right 
```
this.scheduledJobFactory.createJobDetail
                    (job, BusinessStreamNotificationJob.class, job.getJobName(), "scheduled-ldap");
```

Why is Ldap Sych need to know about BusinessStreamNotificationJob class?

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/DashboardJob.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package org.ikasan.dashboard.schedule;
+
+import org.quartz.Job;
+
+public interface DashboardJob extends Job {
```
Comment (andrzej-majewski) [inline ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/DashboardJob.java:5]: should this interface have  some default implementation.

Should BusinessStreamNotificationJob be extending default imple of  DashboardJob

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/AbstractDashboardSchedulerService.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        (AbstractDashboardSchedulerService.class);
+
+    /**
+     * Scheduler
+     */
+    private Scheduler scheduler;
+
+    protected ScheduledJobFactory scheduledJobFactory;
+
+    protected List<JobDetail> jobDetails;
```
Comment (andrzej-majewski) [inline ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/AbstractDashboardSchedulerService.java]: Is this actually used ?

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/DashboardJob.java (hunk 2 above)
Comment (mick-stewart73) [reply ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/DashboardJob.java:5]: i am happy to leave this as an interface that all dashboard scheduled tasks implement.

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/AbstractDashboardSchedulerService.java (hunk 3 above)
Comment (mick-stewart73) [reply ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/schedule/AbstractDashboardSchedulerService.java]: removed

File: ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/security/schedule/LdapDirectorySynchronisationSchedulerService.java (hunk 1 above)
Comment (mick-stewart73) [reply ikasaneip/visualisation/dashboard/src/main/java/org/ikasan/dashboard/security/schedule/LdapDirectorySynchronisationSchedulerService.java]: sorted

**Pull request closed without merging**